ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java

```diff
@@ -678,6 +689,58 @@ public void getGetOptionTest() throws Exception {
     }
   }
 
+  /**
+   * Tests {@link RestUtils#getHeader(Map, String, boolean)}.
+   * @throws RestServiceException
+   */
+  @Test
+  public void getHeaderTest() throws RestServiceException {
+    Map<String, Object> args = new HashMap<>();
+    args.put("HeaderA", "ValueA");
+    args.put("HeaderB", null);
+
+    // get HeaderA
+    assertEquals("Header value does not match", args.get("HeaderA"), RestUtils.getHeader(args, "HeaderA", true));
+    assertEquals("Header value does not match", args.get("HeaderA"), RestUtils.getHeader(args, "HeaderA", false));
+    // get HeaderB
+    assertNull("There should be no value for HeaderB", RestUtils.getHeader(args, "HeaderB", false));
+    try {
+      RestUtils.getHeader(args, "HeaderB", true);
+      fail("Getting HeaderB as required should have failed");
+    } catch (RestServiceException e) {
+      assertEquals("Unexpected RestServiceErrorCode", RestServiceErrorCode.InvalidArgs, e.getErrorCode());
+    }
+
+    // get HeaderC
+    assertNull("There should be no value for HeaderC", RestUtils.getHeader(args, "HeaderB", false));
+    try {
+      RestUtils.getHeader(args, "Headerc", true);
+      fail("Getting HeaderB as required should have failed");
+    } catch (RestServiceException e) {
+      assertEquals("Unexpected RestServiceErrorCode", RestServiceErrorCode.MissingArgs, e.getErrorCode());
+    }
+  }
+
+  /**
+   * Tests for {@link RestUtils#getLongHeader(Map, String, boolean)}.
+   * @throws RestServiceException
+   */
+  @Test
+  public void getLongHeaderTest() throws RestServiceException {
+    Map<String, Object> args = new HashMap<>();
+    args.put("HeaderA", 1000L);
```

Comment:
add a test case where you put a stringified long as the arg value, like "1000".
also test that case mentioned above where an arg is missing but not required.

Reply:
Done